provider/googleProvider.js

```diff
@@ -92,6 +92,9 @@ class GoogleProvider {
           },
           additionalProperties: false,
         },
+        cloudFunctionSecretEnvironmentVariables: {
+          type: 'object',
```

Comment:
In experimenting with this it turns out that the expect structure Google expects is something like this:

```
{
   'secretEnvironmentVariables': [
     {
        'key': 'SOME_ENV_VAR',
        'secret': 'secret-manager-id',
        'version': 'latest',  // Can be 'latest' or a number
        'project': 'project-id' // Optional, defaults to project of the function
     }
   ]
}
```

What would be the best way to represent this with validations here?

Reply:
these validations are intended for config in serverless configuration - it can be transformed internally to better represent what Google expects

Reply:
The current proposal would allow passing only a single value - what, in your opinion, would be the best way to represent these settings in serverless configuration?

Reply:
In the current form, it would be possible to define at most a single secret variable - how do you think it would be best to expose these settings in serverless configuration?

Reply:
Sorry based on the names of these properties, I assumed these validations are validating the config for the output to Deployment Manager. Eg `cloudFunctionEnvironmentVariables` is exactly the name that Deployment Manager expects and `secretEnvironmentVariables` is also what Deployment Manager expects.

The config on the serverless side I've explained below.

Reply:
In terms of the array, I assumed `object` would be sufficient as I'm not across how we would configure an array validation with this?

Reply:
From the proposed syntax that you've mentioned:
```
provider:
  secrets:
  - key: 'API_KEY'
    secret: 'secret-id'
    version: 'latest'
    projectId: 'project-id' # Optional, defaults to function project
```

it won't work, as the proposed syntax is an array of objects that have `key`, `secret`, `version` and `projectId` properties. I think we could potentially simplify this to be like this:
```
provider:
  secrets:
    'API_KEY':
      secret: 'secret-id'
      version: 'latest'
      projectId: 'project-id' # Optional, defaults to function project
```

so the `secrets` would be an object where key is the key of the secret and the value are the rest of properties. I think it would require change in how it's being passed to the Deployment Manager. 

What do you think? Am I missing something here?

Reply:
@pgrzesik I missed this last comment. I'm ok with this approach with the key itself being defined this way. I'll update this PR.

Reply:
Thank you @christophersjchow - please let me know when it's ready for a re-review 🙇

Reply:
I've updated the PR to now support this new format.